### DISCUSSION TITLE
Why do I have to use “id_token_hint” to customize logout redirec_to

### DISCUSSION THREAD

**Discussion by Csirme** [ory/hydra#2925] [Q&A] [answered]
when i use like this and i want to:
oauth2/sessions/logout?client_id=sso-interal-******&post_logout_redirect_uri=https://portal-qa.****.com

the error will show:
Logout+failed+because+query+parameter+post_logout_redirect_uri+is+set+but+id_token_hint+is+missing.

oryd/hydra    v1.10.7

**Comment by vinckr** [comment, marked as answer]
This is what the `id_token_hint` is on a high level.

> ID Token being passed as a hint about the End-User's current or past authenticated session with the Client. 

It is required when you use a `post_logout_redirect_uri`.


>If the URL query contains an ID Token issued by ORY Hydra as the id_token_hint (/oauth2/sessions/logout?id_token_hint=...) then:

> The URL query MAY contain key post_logout_redirect_uri indicating where the user agent should be redirected after the logout completed successfully. Each OAuth 2.0 Client can whitelist a list of URIs that can be used as the value using the post_logout_redirect_uris metadata field: /oauth2/sessions/logout?id_token_hint=...&post_logout_redirect_uri=https://i-must-be-whitelisted/

more reading: https://www.ory.sh/hydra/docs/concepts/logout#logout-flow

**Reply by Csirme** [reply]
I've read this, but it's not what I want. I don't want to redirect through idtoken + redirect uri
